project/api/views.py

```diff
@@ -125,6 +129,111 @@ class SpacetimeVolumeViewSet(viewsets.ModelViewSet):
+
+        print(
+            ">🗺️ STV Filename: {}, Size: {}MB".format(
+                request.data["territory"],
+                round(request.data["territory"].size / 1048576, 2),
+            )
+        )
+
+        content = request.data["territory"].read().decode("utf-8")
+        geom = GEOSGeometry(content)
+        if geom.srid != 4326:
+            print(
+                "Invalid SRID = {}, in file {}".format(
+                    geom.srid, request.data["territory"]
+                )
+            )
+            raise ValidationError("Geometry SRID must be 4326")
+        start_date = float(request.data["start_date"])
+        end_date = float(request.data["end_date"])
+
+        def _overlaps():
+            """
+            Return overlapping STVs
+            """
+            return SpacetimeVolume.objects.raw(
+                """
+                SELECT id, entity_id, end_date, start_date, ST_Union(xing) FROM (
+                    SELECT *,
+                        (ST_Dump(ST_Intersection(territory, diff))).geom as xing
+                    FROM (
+                        SELECT *,
+                            ST_Difference(
+                                territory,
+                                %(geom)s::geometry
+                            ) as diff
+                        FROM (
+                            SELECT *
+                            FROM api_spacetimevolume as stv
+                            WHERE stv.end_date >= %(start_date)s::numeric(10,1)
+                                AND stv.start_date <= %(end_date)s::numeric(10,1)
+                                AND ST_Intersects(
+                                    territory,
+                                %(geom)s::geometry)
+                        ) as foo
+                    ) as foo
+                ) as foo
+                WHERE ST_Dimension(xing) = 2 AND ST_Area(xing) > 10
+                GROUP BY id, entity_id, start_date, end_date
+                """,
+                {"geom": geom.ewkt, "start_date": start_date, "end_date": end_date},
+            )
+
+        if "cacheops" in settings.INSTALLED_APPS:
+            # Cache overlaps query in production
+            _overlaps = (
+                cached_as(
+                    SpacetimeVolume.objects.filter(
+                        territory__overlaps=geom,
+                        start_date__lte=end_date,
+                        end_date__gte=start_date,
+                    ),
+                    extra=(start_date, end_date),
+                )
+            )(_overlaps)
+
+        overlaps_db = _overlaps()
+        overlaps = []
+        if "overlaps" in request.data:
+            for i in overlaps_db:
+                if not str(i.pk) in request.data["overlaps"]:
+                    overlaps.append(i.pk)
+            if overlaps:
+                raise ValidationError(
+                    ('{"unsolved overlap": %(values)s}'), params={"values": overlaps}
```

Comment:
This will not return any valid information for the frontend

Reply:
Wdym? It seems to be working in dev environment for me

Reply:
Somehow I am receiving a debug error page instead of JSON response.